### DISCUSSION TITLE
How do I "Materialise All" for assets that are in different code locations.

### DISCUSSION THREAD

**Discussion by bolinzzz** [dagster-io/dagster#19184] [Q&A] [answered]
Hi team, on a very high level, I have the following setup. I have two code locations A and B. 

Code location A consists extract and load python assets (both partitioned and non partitioned).
Code location B consists of dbt assets, forming a network dependent on assets from code location A.

I wanted to trigger a materialise all, for all assets in a lineage from code location A. See screenshot:
![Screenshot 2024-01-12 at 12 29 59 PM](https://github.com/dagster-io/dagster/assets/101250079/3d6683f0-d766-4440-b6f0-9d22354e5eef)

However, I am prompt with 
![Screenshot 2024-01-12 at 12 27 24 PM](https://github.com/dagster-io/dagster/assets/101250079/76ddce98-eea2-4532-a6fa-d38fc2111b2f)

Seems like I need to form a connect graph in which root assets shared the same partitioning. How do I do that? In my pipeline, it is moving from partitioned extract and load python assets to non partitioned transformation dbt assets.

**Comment by OwenKephart** [comment, marked as answer]
Hi @bolinzzz, unfortunately this is not currently possible. That error text is confusing to unwind, but the correct parsing is:

"defined in the same code location" and ("share a partition space" or "form a connected graph in which root assets share the same partitioning")

**Reply by bolinzzz** [reply]
Thanks @OwenKephart! In that case, will it be possible to create a job that targets an arbitrary set of dbt assets during run time based on the `dbt_select` expression passed into the run config? 

Have you seen anyone doing this before? We will like to have the ability to bulk select and full-refresh all downstream of a particular source, something like this 
```
$ dbt run --select source:jaffle_shop.orders+ --full-refresh
```
The source resides in another code location, hence I can't just materialise all via the dagster UI

**Reply by jamiedemaria** [reply]
dynamically creating the contents of a job based on run config isn't really something we support. If you're running jobs with the same sets of dbt_assets repeatedly, though you could make an asset job using `define_asset_job` and the asset selection syntax https://docs.dagster.io/concepts/assets/asset-selection-syntax

**Reply by bolinzzz** [reply]
Thank you @jamiedemaria, that's exactly what I am doing at the moment. Is the support for "Materialise All" across different code locations part of the roadmap?

**Reply by jamiedemaria** [reply]
I don't believe so

**Comment by Giangblackk** [comment]
@bolinzzz I have the same problem as you, and I found out that Dagster has recently added a feature called [Declarative Automation](https://docs.dagster.io/concepts/automation/declarative-automation#declarative-automation) which allow developers to automatically materialize downstream assets even though they are defined in different code locations (that is how I interpret it, but it's propably more useful than that).
Here is the video introducing this feature. [https://www.youtube.com/watch?v=Z77s50b_Sks](https://www.youtube.com/watch?v=Z77s50b_Sks)
I tried and it works to me. Hope it helps you.
